templates/taxi/car_detail.html

```diff
@@ -3,20 +3,45 @@
 {% block content %}
   <h1>
     {{ car.model }}
-    <a href="{% url 'taxi:car-delete' pk=car.id %}" class="btn btn-danger link-to-page">
-      Delete
-    </a>
-
-    <a href="{% url 'taxi:car-update' pk=car.id %}" class="btn btn-secondary link-to-page">
-      Update
-    </a>
   </h1>
   <p>Manufacturer: ({{ car.manufacturer.name }}, {{ car.manufacturer.country }})</p>
-  <h1>Drivers</h1>
+  
+   <div class="d-flex justify-content-between align-items-center">
+    <h1>Drivers</h1>
+    {% if user.is_authenticated %}
+      {% if user in car.drivers.all %}
+        <form action="{% url 'taxi:remove_driver' car.id user.id %}" method="post">
+            {% csrf_token %}
+            <button type="submit" class="btn btn-danger">Delete me from this car</button>
+        </form>
+      {% else %}
+         <form action="{% url 'taxi:assign_driver' car.id %}" method="post">
+            {% csrf_token %}
+            <button type="submit" class="btn btn-success">Assign me to this car</button>
+        </form>
+      {% endif %}
+    {% endif %}
+  </div>
+  
   <hr>
   <ul>
-    {% for driver in car.drivers.all %}
-      <li>{{ driver.username }} ({{ driver.first_name }} {{ driver.last_name }})</li>
-    {% endfor %}
+    {% if car.drivers.all %}
+      {% for driver in car.drivers.all %}
+        <li>{{ driver.username }} ({{ driver.first_name }} {{ driver.last_name }})</li>
+      {% endfor %}
+    {% else %}
+      <p>This car doesn't have any drivers yet.</p>
+    {% endif %}
   </ul>
+  
+  <div class="d-flex justify-content-start">
+   <a href="{% url 'taxi:car_delete' pk=car.id %}" class="btn btn-danger link-to-page">
+      Delete
+    </a>
+
+    <a href="{% url 'taxi:car_update' pk=car.id %}" class="btn btn-secondary link-to-page">
```

Comment:
The URL identification parameter `name` in the `url` function should use `-` to split words, not `_`. It should be 'taxi:car-update' instead of 'taxi:car_update'.